phoenix-core/src/main/java/org/apache/phoenix/compile/ColumnProjector.java

```diff
@@ -36,11 +36,17 @@
  */
 public interface ColumnProjector {
     /**
-     * Get the column name as it was referenced in the query
```

Comment:
Changing the semantics for getName() makes sense.

My concerns are:
- Does connectors or PQS refer it , and if yes is that a pronlem?
- Does this have backwards compatibility implications ?

My gut feeling is that it's never used outside the client context, but I will check this.

Reply:
PQS and Connectors do not refer this class.

Reply:
We don't serialize ColumnProjectors, so we're good wrt backwards compatibility as well.